src/Listener.php

```diff
@@ -148,17 +176,31 @@ public function process()
                     APILogger::addInfo('HoldRequestResult', $holdRequestResult);
 
                     if ($holdRequestResult->isSuccess() === false) {
-                        // TODO: Fetch e-mail from patron record
-                        APILogger::addInfo('E-mail time', $holdRequestResult->getHoldRequest()->getDocDeliveryData()->getEmailAddress());
+                        $holdRequest = HoldRequestClient::getHoldRequestById($holdRequestResult->getHoldRequestId());
+                        APILogger::addInfo('HoldRequest', $holdRequest);
 
-                        $patron = PatronClient::getPatronById($holdRequestResult->getHoldRequest()->getPatron());
+                        $patron = PatronClient::getPatronById($holdRequest->getPatron());
 
-                        APILogger::addInfo('Patron', $patron->getId());
+                        APILogger::addInfo('Patron', $patron);
                         APILogger::addInfo('E-mail', $patron->getEmails());
                         APILogger::addInfo('BarCodes', $patron->getBarCodes());
-                        // TODO: Send e-mail to patron notifying failure
```

Comment:
I'm rather bothered by logging this type of personally identifiable information. Is there a way we could hash these values if we need to log events that track a user's activity? Is logging of this information against our policies?

I have a similar question about how long we keep a user's holds information. Are we expunging it after the checkin back to ReCAP? or once the ILS system takes over?